src/Board.cpp

```diff
@@ -0,0 +1,226 @@
+ * @param other The board to copy.
+ */
+Board::Board(const Board& other) {
+    for (const auto& pair : other.m_board) {
+        m_board.push_back({ pair.first->clone(), pair.second });
+    }
+    m_turn = other.m_turn;
+}
+
+/**
+ * @brief Creates a Piece object based on the given character.
+ * @param piece The character representing the piece to create.
+ * @return A shared pointer to the created Piece object, or nullptr if the character is not recognized.
+ */
+std::shared_ptr<Piece> Board::createPiece(const char piece) const
+{
+    static std::map<char, std::function<std::shared_ptr<Piece>()>> piece_map = {
+        {'P', []() { return std::make_shared<Pawn>(PieceColor::WHITE); }},
+        {'p', []() { return std::make_shared<Pawn>(PieceColor::BLACK); }},
+        {'R', []() { return std::make_shared<Rook>(PieceColor::WHITE); }},
+        {'r', []() { return std::make_shared<Rook>(PieceColor::BLACK); }},
+        {'N', []() { return std::make_shared<Knight>(PieceColor::WHITE); }},
+        {'n', []() { return std::make_shared<Knight>(PieceColor::BLACK); }},
+        {'B', []() { return std::make_shared<Bishop>(PieceColor::WHITE); }},
+        {'b', []() { return std::make_shared<Bishop>(PieceColor::BLACK); }},
+        {'Q', []() { return std::make_shared<Queen>(PieceColor::WHITE); }},
+        {'q', []() { return std::make_shared<Queen>(PieceColor::BLACK); }},
+        {'K', []() { return std::make_shared<King>(PieceColor::WHITE); }},
+        {'k', []() { return std::make_shared<King>(PieceColor::BLACK); }}
+    };
+
+    auto it = piece_map.find(piece);
+    if (it != piece_map.end()) {
+        auto aPiece = it->second();
+        return aPiece;
+    }
+
+    return nullptr;
+}
+
+/**
+ * @brief Executes the given command string, moving a piece from one position to another.
+ * @param command The command string, containing the source and destination positions.
+ * @return An integer code representing the result of the move (e.g., illegal move, check, etc.).
+ */
+int Board::runCommand(const std::string& command)
+{
+    if (!isValidInput(command))
+        return 11;
+    
+    Place src = { command[0] - 'a', command[1] - '1' };
+    Place dst = { command[2] - 'a', command[3] - '1' };
+
+    auto itSrc = std::find_if(m_board.begin(), m_board.end(),
+        [&](const auto& p) {return p.second == src; });
+    auto itDst = std::find_if(m_board.begin(), m_board.end(),
+        [&](const auto& p) {return p.second == dst; });
+
+    // checking if there is not piece at the source  
+    if (itSrc == m_board.end())
+        return 11;
+    
+    // checking if src color is the one who has to play
+    else if (itSrc->first->getColor() != m_turn)
+        return 12;
+
+    // checking if dst color is not the same as the one who is playing
+    if (itDst != m_board.end() && itDst->first->getColor() == m_turn)
+        return 13;
+
+    bool emptyDst = itDst == m_board.end() ? true : false;

```

Comment:
The `?` operator is redundant. You could just do `bool emptyDst = itDst == m_board.end()`